src/main/java/com/notably/commons/core/path/RelativePath.java

```diff
@@ -1,53 +1,112 @@
+        this.paths = new ArrayList<>();
+        for (String obj : relativePathString.split("/")) {
+            if (!obj.trim().equals("")) {
+                this.paths.add(obj.trim());
+            }
+        }
+    }
+
     /**
-     * TODO: Add Javadoc
+     * Check if string is a valid relative path.
+     * @param relativePathString String used to check validity.
+     * @return returns true if String is a valid relative Path.
      */
     public static boolean isValidRelativePath(String relativePathString) {
-        // TODO: Add implementation
-        return false;
+        return relativePathString.matches(VALIDATION_REGEX);
     }
 
     /**
-     * TODO: Add Javadoc
+     * Instantiate a RelativePath object if the string is valid.
+     * @param relativePathString used to create RelativePath.
+     * @return the converted RelativePath object.
+     * @throws InvalidPathException if the String is invalid (Not relative path).
      */
     public static RelativePath fromString(String relativePathString) throws InvalidPathException {
-        // TODO: Add implementation
-        return new RelativePath();
+        if (!isValidRelativePath(relativePathString)) {
+            throw new InvalidPathException("Not a relative path");
+        }
+        return new RelativePath(relativePathString);
+
     }
 
     /**
-     * TODO: Add Javadoc
+     * Convert from absolute path to relative path.
+     * @param absolutePath to convert to relative path.
+     * @param currentWorkingPath of the current working directory.
+     * @return the converted relative path.
+     * @throws InvalidPathException
      */
     public static RelativePath fromAbsolutePath(AbsolutePath absolutePath,
             AbsolutePath currentWorkingPath) throws InvalidPathException {
-        // TODO: Add implementation
-        return new RelativePath();
+        return absolutePath.toRelativePath(currentWorkingPath);
+
     }
 
     /**
-     * TODO: Add Javadoc
+     * Convert relative path to absolute path.
+     * @param currentWorkingPath of the current working directory.
+     * @return the absolute path of the relative path.
      */
-    public AbsolutePath toAbsolutePath(AbsolutePath currentWorkingPath) {
-        // TODO: Add implementation
-        return new AbsolutePath();
+    public AbsolutePath toAbsolutePath(AbsolutePath currentWorkingPath) throws InvalidPathException {
+        List<String> temp = new ArrayList<>(this.paths);
+        List<String> temp2 = new ArrayList<>(currentWorkingPath.getComponents());
+        for (String obj : temp) {
+            if (obj.equals("..")) {
+                if (temp2.remove(temp2.size() - 1) == null) {
+                    throw new InvalidPathException("Root block does not have a parent");
+                }
+            } else if (obj.equals(".")) {
+                // Do nothing
+            } else {
+                temp2.add(obj);
+            }
+        }
+        return new AbsolutePath(temp2);
     }
```

Comment:
You should be able to just call `AbsolutePath.fromRelativePath(this, currentWorkingPath)` here!